src/inmanta/server/services/compilerservice.py

```diff
@@ -478,30 +478,34 @@ async def is_compiling(self, environment_id: uuid.UUID) -> Apireturn:
             return 200
         return 204
 
-    async def _calculate_recompile_wait(self, compile: data.Compile):
-        last_run = await data.Compile.get_last_run(compile.environment)
+    def _calculate_recompile_wait(
+        self, compile_requested: datetime.datetime, last_compile_completed: datetime.datetime, now: datetime.datetime
+    ):
         wait_time = opt.server_autrecompile_wait.get()
-        if not last_run or wait_time == 0:
+        if wait_time == 0:
             wait: float = 0
         else:
-            assert last_run.completed is not None
-            if last_run.completed >= compile.requested:
-                wait = max(0, wait_time - (last_run.completed - compile.requested).total_seconds())
+            assert last_compile_completed is not None
```

Comment:
This assert makes little sense from a typing standpoint (this method expects `datetime.datetime`, not `Optional[datetime.datetime]`. It would make more sense to do this assertion in the caller, right before calling this method. I would guess mypy would actually complains about this.